### ISSUE TITLE
Add setting on Combined commission to set how to appply it

### ISSUE THREAD

**Issue by Facyla** [weDevsOfficial/dokan#940]
When marketplace sells many low-price products, the way the combined commission currently works becomes a dissuasive issue for vendors : currently the flat commission applies to every product in a single order, which reduces the vendor's gain, thus preventing them for selling cheap products. 

Example : 
- Combined commission set to $0,50 + 10%
- On a $1 product, commission is $0,60 and vendor earning 0,40 $.   Commission on a total $2 order is $1,20, and vendor eearning $0,80.
- If 2 are sold in a single order, that's $0,80 earning - 2 * 0,50 + 10% of total). 
- Now if vendors sells a pack of 2 (as a single product) instead, flat commission is applied once (single product) : commission becomes 0,70 €, and vendor earning $1,30.

We feel that the commission should be the same whether the vendors sells several units of a product in a single order, or one product that has the same products packaged in a single product, as long as it is in a single order and the consumer price is the same. 
Basically, there are many use case where commission should apply to orders rather than to single products. 

Feature suggested : add an option to set how to apply the flat commission : 
1. Per Product (current behaviour)
2. Per Vendor (per Sub-order)
3. Per Order
Note : (2) and (3) are useful to handle the multi-vendors orders (sub-orders).

**Comment by alamgircsebd** [comment]
There are different types of vendor commissions available in Dokan Plugin. They are,

#1 Global Commission
#2 Vendor Wise Commission
#3 Category Wise Commission
#4 Product Wise Commission

and commission priority wise working: Product Wise=> Category Wise => Vendor Wise => and Lastly Global Commission 

so you can set where want to do the commissions

Check this also: https://wedevs.com/docs/dokan/tutorials/how-to-setup-dokan-vendor-commission/

**Comment by Facyla** [comment]
Thanks for looking into this. 
I suppose feature suggestion was rejected, as this link to the documentation shows no change at all, nor any related commit. 

Please let me explain in more detail our issue and suggestion : 

It seems the setting a fixed global commission applies to each product type, ie.: 
- global mixed commission of 10% + 1$
- 1 product at 10$ => commission is 0.1*10 + 1 = 2$
- 3 different products (each 10$) => commission is 0.1*3*10 + 3*1 = 6$

What we would like to achieve would be : 
- 3 different products (each 10$) => commission would be with fixed commission applied only once, ie. 0.1*3*10 + 1 = 4$

=> Is that possible with the current behaviour ?

Also, even if it is possible, it would require setting custom fixed commission for each and every vendor, which can really become an hassle with numerous vendors. 

The feature suggestion would be to allow setting the fixed commission scope at a global level, without making custom changes  at the other levels : once per product type (current behaviour), or once per order, or once per sub-order (for multi-vendors orders).
